pkg/system/phase2_creating.go

```diff
@@ -152,6 +157,101 @@ func (r *Reconciler) ReconcilePhaseCreatingForMainClusters() error {
 	return nil
 }
 
+func (r *Reconciler) UpgradeDbPostgres16() error {
+	var err error = nil
+
+	// Check the desired image, if it is not as same as db image to upgrade,
+	// don't consider for upgrade
+	// Note: Intention of the below call is to fetch the desired image from Noobaa CR or env var
+	// Passing empty Arg "" to GetDesiredDBImage() in the below case
+	desiredImage := GetDesiredDBImage(r.NooBaa, "")
+	if desiredImage != options.DBImage {
```

Comment:
I don't understand this condition. `options.DBImage` is hardcoded to the upstream DB image.

Reply:
Yes, on ds, it will be [registry.redhat.io/rhel9/postgresql-16](http://registry.redhat.io/rhel9/postgresql-16)

Reply:
Apologies. I thought we will go for whatever is there in options.DBImage for upgrade. Got clarity that the we have to match the noobaa state and use the image indiacated in the noobaa CR. Updating the PR according to that